src/lib/auth.ts

```diff
@@ -0,0 +1,110 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import { Portal } from './types';
+import { COMMANDS } from './constants';
+
+const { findConfig, loadConfig, validateConfig } = require('@hubspot/cli-lib');
+const { setConfig, setConfigPath } = require('@hubspot/cli-lib/lib/config');
+
+let hubspotConfigWatcher: fs.FSWatcher | null;
+let configDependentFeaturesLoaded: boolean = false;
+let loadConfigDependentFeatures: Function;
+let updateConfigDependentFeatures: Function;
+
+const onLoadPath = (configPath: string) => {
+  vscode.commands.executeCommand(
+    'setContext',
+    'hubspot.configPath',
+    configPath
+  );
+  if (!configPath) {
+    vscode.commands.executeCommand(COMMANDS.CONFIG_SET_DEFAULT_ACCOUNT, null);
+    setConfig(null);
+    setConfigPath(null);
+  }
+};
+
+const onLoadHubspotConfig = () => {
+  updateConfigDependentFeatures();
+};
+
+export const loadHubspotConfigFile = (rootPath: string) => {
+  if (!rootPath) {
+    return;
+  }
+
+  console.log(`Root path: ${rootPath}`);
+
+  const path = findConfig(rootPath);
+  onLoadPath(path);
+
+  console.log(`Path: ${path}`);
+
+  if (!path) {
+    return;
+  }
+
+  const config = loadConfig(path);
+
+  console.log(
+    'loadedHubspotConfig',
+    config.defaultPortal,
+    JSON.stringify(config.portals.map((p: Portal) => p.name))
+  );
+
+  if (!validateConfig()) {
+    throw new Error(`Invalid config could not be loaded: ${path}`);
+  } else {
+    onLoadHubspotConfig();
```

Comment:
Could we just call `updateConfigDependentFeatures()` directly here?

Reply:
Yes, I had some different code in `onLoadHubSpotConfig` before that was removed, but left it for later use, but it can be removed now.